### ISSUE TITLE
ERROR in The Angular Compiler requires TypeScript >=3.6.4 and <3.9.0 but 3.9.3 was found instead

### ISSUE THREAD

**Issue by ErVishalM** [angular/angular#37342]
### **Description**
I just upgraded to Angular/cli 9.1.7 and some other packages also using `ng update`

Old packags.json

```
"dependencies": {
   "@angular/animations": "^9.1.7",
       "@angular/common": "^9.1.7",
       "@angular/compiler": "^9.1.7",
       "@angular/core": "^9.1.7",
       "@angular/forms": "^9.1.7",
       "@angular/platform-browser": "^9.1.7",
       "@angular/platform-browser-dynamic": "^9.1.7",
       "@angular/router": "^9.1.7",
   "moment": "^2.25.3",
   },
   "devDependencies": {
       "@angular/cli": "^9.1.6",
       "@angular/compiler-cli": "^9.1.7",
       "@angular/language-service": "^9.1.7",
       "@types/node": "^12.12.39",
       "karma": "^5.0.8",
       "typescript": "^3.7.5"
   }
```
new package.json

```
"dependencies": {
	"@angular/animations": "^9.1.9",
        "@angular/common": "^9.1.9",
        "@angular/compiler": "^9.1.9",
        "@angular/core": "^9.1.9",
        "@angular/forms": "^9.1.9",
        "@angular/platform-browser": "^9.1.9",
        "@angular/platform-browser-dynamic": "^9.1.9",
        "@angular/router": "^9.1.9",
        "moment": "^2.26.0",
    },
    "devDependencies": {
        "@angular/cli": "^9.1.7",
        "@angular/compiler-cli": "^9.1.9",
        "@angular/language-service": "^9.1.9",
        "@types/node": "^12.12.42",
        "karma": "^5.0.9",
        "typescript": "^3.9.3"
    }

```
Now when I run `ng s`, I get an error,
**`ERROR in The Angular Compiler requires TypeScript >=3.6.4 and <3.9.0 but 3.9.3 was found instead`**

**Why did Angular update me to Typescript 3.9.3 if it's not supported?**
Error SS
<img width="647" alt="82731678-3616bf80-9d26-11ea-95dd-bccab299eeee" src="https://user-images.githubusercontent.com/43669644/83221892-bad56380-a194-11ea-8f9f-b4e27544a009.png">

Now i have updated all the other packages and downgrade typescript to old version but still its showing me to update
![image](https://user-images.githubusercontent.com/43669644/83221936-da6c8c00-a194-11ea-888a-d70d8ba875b9.png)

**Comment by JoostK** [comment]
Your original `package.json` specifies that it wants TypeScript `^3.7.5`, so that includes the recently released TS 3.9 version which is not supported in Angular 9.1. This is why `npm outdated` allows you to upgrade to said version. Since TypeScript does not follow semver with its releases, and Angular only supports specific versions, you should change the version range of TypeScript to use a `~` instead of a `^` to constrain updates to patch releases only.

**Comment by ErVishalM** [comment]
Thank you so much @JoostK  for this information. I didn't have that knowledge 👍
